src/main.rs

```diff
@@ -120,79 +120,48 @@ fn main() {
-                &mut section_counts,
-                &section_regex,
-                &mut section_text,
-                &mut law_locations,
-                &mut bill_locations,
-            );
-            println!("Total sections: {}", section_counts.total);
-            println!("Amending sections: {}", section_counts.amending);
-            println!(
-                "Amending sections by striking and inserting: {}",
-                section_counts.amending_by_striking_and_inserting
-            );
-            println!(
-                "Amending sections by striking: {}",
-                section_counts.amending_by_striking
-            );
-            println!(
-                "Amending sections by inserting: {}",
-                section_counts.amending_by_inserting
-            );
-            println!("Repealing sections: {}", section_counts.repealing);
-            println!("Other sections: {}", section_counts.other);
-
-            // Get amended or repealed law sections concurrently
-            let (tx, rx) = mpsc::channel();
-
-            // Remove a law location from the collection to use last
-            let law_location = law_locations.iter().next().unwrap().clone();
-            assert!(
-                law_locations.remove(&law_location),
-                "Cannot remove law location"
-            );
-            for law_location in law_locations {
-                for law_location_section in law_location.sections {
-                    get_law_section(&law_location.chapter, &law_location_section, tx.clone());
+            let bill = collect_bill_sections(text_nodes, &section_regex);
+            // Count type of bill sections with regex
+            let _section_counts = count_bill_section_types(&bill, &section_regex);
+
+            // Iterate through bill to get list of all needed sections for downloading
+            let mut required_law_sections: Vec<(String, String)> = Vec::new();
+            for bill_section in &bill {
+                for section in &bill_section.law_sections.section_numbers {
+                    required_law_sections.push((
+                        bill_section.law_sections.chapter_number.clone(),
+                        section.to_string(),
+                    ))
                 }
             }
+            // Remove duplicates
+            required_law_sections.sort();
+            required_law_sections.dedup();
 
-            // Pop a vector of law location sections to use last
-            let mut law_location_sections = law_location.sections.clone();
-            let law_location_section = law_location_sections.pop().unwrap();
-            for law_location_section in law_location_sections {
-                get_law_section(&law_location.chapter, &law_location_section, tx.clone());
+            // Download required law sections concurrently
+            let (tx, rx) = mpsc::channel();
+            let chapter_section = required_law_sections.pop().unwrap();
```

Comment:
Can we write this?
let (chapter, section) = required_law_sections.pop().unwrap();

Same if for.